doc/release/1.2.0-notes.rst

```diff
@@ -6,28 +6,91 @@ SciPy 1.2.0 Release Notes
+This release requires Python 2.7 or 3.4+ and NumPy 1.8.2 or greater.
 
+For running on PyPy, PyPy3 6.0+ and Numpy 1.15.0 are required.
+
+Release highlights:
+
+   -
 
 New features
 ============
 
+`scipy.cluster` improvements
+----------------------------
+
+`scipy.cluster.vq.kmeans2` gained a new initialization method, kmeans++.
+
 `scipy.special` improvements
 ----------------------------
 
 The function `softmax` was added to `scipy.special`.
 
+`scipy.optimize` improvements
+-----------------------------
+
+The one-dimensional nonlinear solvers have been given a unified interface `scipy.optimize.root_scalar`, similar to the `scipy.optimize.root` interface for multi-dimensional solvers. `scipy.optimize.root_scalar(f, bracket=[a ,b], method="brenth")` is equivalent to `scipy.optimize.brenth(f, a ,b)`.  If no `method` is specified, an appropriate one will be selected based upon the bracket and the number of derivatives available.
+
+The so-called Algorithm 748 of Alefeld, Potra and Shi for root-finding within an enclosing interval has been added as `scipy.optimize.toms748`. This provides guaranteed convergence to a root with convergence rate per function evaluation of approximately 1.65 (for sufficiently nice functions.)
+
+`differential_evolution` now has the `updating` and `workers` keywords. The first chooses between continuous updating of the best solution vector (the default), or once per generation. Continuous updating can lead to faster convergence. The `workers` keyword accepts an `int` or map-like callable, and parallelises the solver (having the side effect of updating once per generation). Supplying an `int` evaluates the trial solutions in N parallel parts. Supplying a map-like callable allows other parallelisation approaches (such as `mpi4py`, or `joblib`) to be used.
+
+`dual_annealing` is a new method which uses two annealing processes to accelerate the convergence towards the global minimum of an objective mathematical function. The first annealing process controls the stochastic Markov chain searching and the second annealing process controls the deterministic minimization.  
+So, dual annealing is a hybrid method that takes advantages of stochastic and local deterministic searching in an efficient way.
+
+`shgo` (simplicial homology global optimization) is a powerful new general purpose global optimization (GO) algorithm appropriate for solving black box and derivative free optimization (DFO) problems. The algorithm is proven to convergence to the global solution in finite time and shows [highly competitive performance compared to cutting edge commercial and open source algorithms available in literature](https://stefan-endres.github.io/files/shgo_jogo.pdf). The convergence holds for non-linear inequality and equality constraints. In addition to returning a global minimum, the algorithm also returns any other global and local minima found after every iteration. This makes it useful for exploring the solutions in a domain.
```

Comment:
* Maybe add something about `newton` can now take either a scalar or an array #8357?
* Also @pvanmulbregt improved Halley to switch to Newton in case convergence is negative #8882, which might change the root slightly from previous answer depending on tolerance.